File: js/hamming.js
Hamming = function() {};

Hamming.prototype.compute = function(strand1, strand2, count=0) {
  if (strand1.length != strand2.length) {throw "DNA strands must be of equal length."};
  for (var i=0; i<strand1.length; i++) {
    if (strand1[i] != strand2[i]) {
      count+=1
    };
  };
  return count
};

module.exports = Hamming;

File: ruby/hamming.rb
class Hamming

  def self.compute(strand1, strand2, distance=0)
    raise ArgumentError, "Strands must be of equal length" if strand1.length != strand2.length
    strand1.split("").each_with_index do |base, index|
      distance+=1 if base != strand2.split("")[index]
    end
    distance
  end
end
